File: src/test/java/handler/OpponentRanksHandlerTest.java
package handler;

import collect.misc.OpponentRanks;
import handler.OpponentRanksHandler;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import java.util.HashMap;
import java.util.Map;

import static org.mockito.Mockito.verify;

class OpponentRanksHandlerTest {

    @Mock
    OpponentRanks opponentRanks;

    @InjectMocks
    OpponentRanksHandler opponentRanksHandler;

    @BeforeEach
    void setUp() {
        MockitoAnnotations.initMocks(this);
    }

    @Test
    void shouldHandleOpponentRanksRequest() {
        Map<String, String> testMap = new HashMap<>();
        testMap.put("sport", "nfl");
        opponentRanksHandler.handleRequest(testMap);
        verify(opponentRanks).getOpponentRanks("nfl");
    }

    @Test
    void shouldHandleOpponentRanksRequestNonNFL() {
        Map<String, String> testMap = new HashMap<>();
        testMap.put("sport", "nba");
        opponentRanksHandler.handleRequest(testMap);
        verify(opponentRanks).getOpponentRanks("nba");
    }
}

File: src/test/kotlin/handler/OpponentRanksHandlerTest.kt
package handler

import io.mockk.every
import io.mockk.spyk
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach

class OpponentRanksHandlerTest {
    private val sport = "a sport"
    private val opponentRanksHandler = spyk<OpponentRanksHandler>()
    private val opponentRanksResult: Map<String, Map<String, Int>> = mapOf()

    @BeforeEach
    fun setUp() {
        every { opponentRanksHandler.getOpponentRanksData(sport) } returns opponentRanksResult
    }

    @Test
    fun `should handle request`() {
        val result = opponentRanksHandler.handleRequest(mapOf("sport" to sport))
        assertEquals(opponentRanksResult, result)
    }
}